### ISSUE TITLE
mod_http_upload Potential PUT_URL bug

### ISSUE THREAD

**Issue by SafwatHalaby** [processone/ejabberd#820]
With the following configuration, http file uploads fail:

nginx:
```
	server {
	    listen *:443 ssl;
            ...Irrelevant things here...

            root /home/www/;
	    
	    client_max_body_size 12m;
	    location / {
	        # Pass all requests to ejabberd, except for GET and HEAD requests.
	        limit_except GET HEAD { 
	            proxy_pass http://localhost:5444;
	        }
	        proxy_set_header Host $host;
	
	        if ($request_method = GET) {
	            add_header X-Frame-Options DENY;
	        }
	    }
```

Ejabberd:
```
    docroot: "/home/www/xmpp"
    put_url: "https://<truncated>.com:443/xmpp/"
    get_url: "https://<truncated>.com:443/xmpp/"
```

However, changing the `put_url` line to the following makes it all works fine:
```
put_url: "https://<truncated>.com:443/"
```

My guess is that with the first scenario, ejabbered attempted to use `/home/www/xmpp/xmpp/`.
This is either a bug or vague documentation. According to the docs:
> docroot: Uploaded files are stored below the directory specified (as an absolute
  path) with this option.  The keyword @HOME@ is replaced with the home
  directory of the user running ejabberd.

In other words, I would have expected that the `put_url` and `get_url` parameters have no effect at all on where the file is stored. Even `put_url: "https://<truncated>.com:443/osakcsa/ckjzc/"` should work.

**Comment by weiss** [comment]
With `put_url: "https://<truncated>.com:443/xmpp/"`, you must configure the `ejabberd_http` listener like this:

    -
      module: ejabberd_http
      # [...]
      request_handlers:
        "xmpp": mod_http_upload

We should probably clarify this in the documentation.

(You can omit the explicit port number from the `put_url` and `get_url` if it's port 443 for HTTPS, BTW.)

**Comment by SafwatHalaby** [comment]
That resolved it. Thanks :)
